### ISSUE TITLE
CA2235 complains that List is not serializeable.

### ISSUE THREAD

**Issue by jmarolf** [dotnet/roslyn-analyzers#944]
#### Repro steps

Consider the code below:

```C#
    public sealed class ProcessOutput : ISerializable
    {
        public ProcessOutput(IEnumerable<string> outputLines)
        {
            OutputLines = outputLines.ToList();
        }

        private ProcessOutput(SerializationInfo info, StreamingContext context)
        {
            OutputLines = info.GetValue(nameof(OutputLines), typeof(List<string>)) as List<string>;
        }

        public List<string> OutputLines { get; }
    }
```

#### Expected behavior
No errors as List<string> is serialize-able
.
#### Actual behavior
Severity | Code | Description | 
------------ | ------------- | ------------- 
Warning	| CA2235 |	Field OutputLines is a member of type ProcessOutput which is serializable but is of type System.Collections.Generic.List<string> which is not serializable

**Comment by mavasani** [comment]
Doesn't repro on latest analyzer packages - https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers/1.2.0-beta1

**Comment by AvishagCs** [comment]
Hi. 
I'm having the same serialization issue as described here.
I downloaded the package you said (Microsoft.CodeAnalysis.FxCopAnalyzers 1.2.0-beta1), but now I can't compile my project and I receive error like:
CSC : error AD0001: Analyzer 'System.Runtime.Analyzers.SpecifyCultureInfoAnalyzer' threw an exception of type 'System.InvalidOperationException' with message 'Feature 'IOperation' is disabled.

When I downloaded the last released package of Microsoft.CodeAnalysis.FxCopAnalyzers, the compilation went well, but the serialization problem wasn't solved.

Any idea?
